### PR TITLE
Fixed #11154, #17904 -- Added creation of proxy model permissions

### PR DESCRIPTION
**Pull request by acatton**
I just updated @charettes' patch in #146, and added release notes.

**Edit:** Tickets links:

 * https://code.djangoproject.com/ticket/11154
 * https://code.djangoproject.com/ticket/17904

### CODE REVIEW

**Comment by charettes** [comment]
Buildbot test this please.

File: django/contrib/auth/management/__init__.py [hunk 1]
```diff
@@ -82,7 +82,8 @@ def create_permissions(app_config, verbosity=2, interactive=True, using=DEFAULT_
     for klass in app_config.get_models():
         # Force looking up the content types in the current database
         # before creating foreign keys to them.
-        ctype = ContentType.objects.db_manager(using).get_for_model(klass)
+        ctype = ContentType.objects.db_manager(using).get_for_model(klass,
+                                                                    for_concrete_models=False)
```
Comment (charettes) [inline django/contrib/auth/management/__init__.py]: `for_concrete_model=False` (this patch is not using `get_for_models` anymore).

File: tests/proxy_models/tests.py [hunk 2]
```diff
@@ -224,6 +224,11 @@ def test_permissions_created(self):
             Permission.objects.get(name="May display users information")
         except Permission.DoesNotExist:
             self.fail("The permission 'May display users information' has not been created")
+        try:
+            Permission.objects.get(content_type__model='myperson',
+                                   codename='add_myperson')
+        except Permission.DoesNotExist:
+            self.fail("There's no permission 'add_myperson' associated with the MyPerson model")
```
Comment (charettes) [inline tests/proxy_models/tests.py]: `"There's no 'add_myperson' permission ...`

**Comment by acatton** [comment]
@charettes Here you go. Thanks for reviewing this.

**Comment by charettes** [comment]
Buildbot test this please.

**Comment by timgraham** [comment]
Could we please add documentation about how permissions on proxy models work are supposed to work ([ticket #22270](https://code.djangoproject.com/ticket/22270)) along with this change?

**Comment by acatton** [comment]
@timgraham I updated the documentation for the default permissions.

**Comment by timgraham** [comment]
I was hoping for a bit more discussion on how the permissions work, not just the fact that they are created. For example, the ticket asks, "Please explain how proxy models and model permissions interact."

**Comment by auvipy** [comment]
@acatton This branch has conflicts that must be resolved

**Comment by acatton** [comment]
Sorry for the delay, @timgraham I tried to add more documentation, but I'm not sure what to explain. Maybe because I know the issue, and I read this code, I don't understand what questions people would have about.

Could you give me a list of questions I should answer in the documentation?

Thanks.

**Comment by timgraham** [comment]
How do permission betweeen normal models and proxy models interact, e.g. how is 
https://code.djangoproject.com/ticket/11154#comment:53 addressed?

By the way, this will need to be targeted at 1.10 now that 1.9 is feature frozen.

**Comment by charettes** [comment]
@acatton you must investigate all the code in Django that relies on model named permissions and make sure this doesn't introduce any security issue.

For example, the admin uses `django.contrib.auth.get_permission_codename` to retrieve the name of a permission for a specific access (e.g. `has_change_permission`). Since proxy model permissions weren't created automatically prior to this patch it means that a proxy model registered to an admin site wasn't functional unless permissions were explicitly defined in it's `_meta.permissions` option.

If we merge this patch, what is going to happen? Are permissions for proxy models going to be automatically created on the next `migrate`? Do users migrating from Django 1.9 will need to create a data migration and call `Permission.objects.create()` for all their applications proxy models?

As @timgraham said it should also be documented that permission for proxy models are not inherited from the concrete model they are proxying. That means that if I have the `app_model_change` permission I don't necessarily have `app_modelproxy_change` one and vice versa.  

There's a lot of unanswered questions here and since this is security related we should be very careful before merging this change.

**Comment by timgraham** [comment]
Hi Antoine, do you think you can do the investigation that Simon noted soon or should we close this pull request for now?

**Comment by acatton** [comment]
@timgraham I don't think I can look into what Simon said since I feel that I'm missing how other people use permissions, and this change might impact many people.

I think you're right Tim, closing this pull request might be the wisest decision. If someone with a better understanding wants to take over, that would be even better.

Thanks for following up @timgraham .

**Comment by timgraham** [comment]
Thanks for the reply. I've made a note of this pull request and the follow up that needs to happen on the related tickets.

**Pull request closed without merging**